tests/uitools/qml_quick/functional_tests/FloorFilterControllerFuncTest/tst_floorfiltercontroller.qml

```diff
@@ -0,0 +1,189 @@
+        var facility = control.floorManager.facilities[0];
+        verify(facility.facilityId);
+        control.setSelectedFacilityId(facility.facilityId);
+        compare(control.selectedSiteId, facility.site.siteId);
+        compare(control.selectedSite.siteId, facility.site.siteId);
+    }
+
+    // singlesite mode on. only 1 site and 1 facility, automatically select them on load.
+    function test_singleSiteAutoSelect() {
+        var control = loadFloorManagerSingle();
+        verify(1, control.floorManager.sites.count);
+        compare(1, control.sites.count);
+        compare(1, control.facilities.count);
+        verify(control.selectedSiteId) // not empty string
+        verify(control.selectedSite);
+        verify(control.selectedFacilityId);
+        verify(control.selectedFacility);
+    }
+
+
+    Component {
+        id: mapComponentMulti
+        Map {
+            item: PortalItem {
+                itemId: "49520a67773842f1858602735ef538b5"
+                portal: Portal {
+                    Component.onCompleted: load()
+                    onErrorChanged: {
+                        if (loadStatus === Enums.LoadStatusFailedToLoad) {
+                            console.log(error.message);
+                        }
+                    }
+                    credential: viewerCredentialIndoors
+                    url: "https://indoors.maps.arcgis.com/"
+                }
+            }
+            onLoadStatusChanged: console.log("loading map", loadStatus)
+        }
+
+    }
+
+    // don't change the visibility of levels when ff is loaded.
+    // load a new map with floormanager. check its levels visibility with the same map loaded in the floorfilter
+    function test_noModifyOriginalVisibilityLevels() {
+        var control = loadFloorManagerMultiple();
+        var map = createTemporaryObject(mapComponentMulti, floorFilterControllerFuncTest);
+        var control = createTemporaryObject(controlMultiComponent, floorFilterControllerFuncTest);
+        map.load();
+        var spy = createTemporaryQmlObject('import QtQuick 2.0; SignalSpy {signalName : "onLoadStatusChanged"}',
+                                           floorFilterControllerFuncTest);
+        spy.target = map;
+        // wait for map to be loaded, then able to set a spy on the floormanager.
+        spy.wait();
+        spy.clear();
+        spy.target = Qt.binding(function() { return map.floorManager});
+        map.floorManager.load();
+        spy.wait();
+        console.log(map.floorManager.loadStatus);
+        if(map.floorManager.loadStatus !== Enums.LoadStatusLoaded)
+            spy.wait();
+        verify(map.floorManager.levels.length > 0);
+        verify(map.floorManager.levels.length === control.floorManager.levels.length);
+        for (var i = 0; i < map.floorManager.levels; ++i) {
+            verify(map.floorManager.levels[i].visible === control.floorManager.levels[i].visible);
+        }
+    }
```

Comment:
I think some of this logic, could be simplified by injecting the SignalSpy into the component directly, and forcing the FloorManager to load after the map has loaded also:

```qml

   Component {
        id: mapComponentMulti
        Map {
            id: map
            item: PortalItem {
                itemId: "49520a67773842f1858602735ef538b5"
                portal: Portal {
                    Component.onCompleted: load()
                    onErrorChanged: {
                        if (loadStatus === Enums.LoadStatusFailedToLoad) {
                            console.error(error.message);
                        }
                    }
                    onLoadStatusChanged: {
                        if (loadStatus === Enums.LoadStatusLoaded) {
                            map.floorManager.load()
                        }
                    }
                    credential: viewerCredentialIndoors
                    url: "https://indoors.maps.arcgis.com/"
                }
            }
            property SignalSpy spy: SignalSpy {
              target: map.floorManager
              signalName: "onLoadStatusChanged"
            }
        }
    }
```